src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -0,0 +1,173 @@
+package fs2.async.mutable
+
+import fs2._
+import fs2.async.AsyncExt.Change
+
+import fs2.async.{immutable, AsyncExt}
+
+
+
+/**
+ * Asynchronous queue interface. Operations are all nonblocking in their
+ * implementations, but may be 'semantically' blocking. For instance,
+ * a queue may have a bound on its size, in which case enqueuing may
+ * block until there is an offsetting dequeue.
+ */
+trait Queue[F[_],A] {
+
+
+  /**
+   * Enqueue one element in this `Queue`.
+   *
+   * Evaluates to `false` if the queue is full, indicating the `a` was not queued up.
+   * Evaluates to `true` if the `a` was queued up successfully.
+   *
+   * @param a `A` to enqueue
+   */
+  def enqueueOne(a: A): F[Boolean]
+
+  /**
+   * Provides a process that dequeue from this queue.
+   * When multiple consumers dequeue from this queue,
+   * they dequeue in first-come, first-serve order.
+   *
+   * Please use `Topic` instead of `Queue` when all subscribers
+   * need to see each value enqueued.
+   *
+   * This process is equivalent to `dequeueBatch(1)`.
+   */
+  def dequeue: Stream[F, A]
```

Comment:
Should we expose a `dequeueOne: F[A]` operation also? I'd actually call it `dequeue1` (analogous to `await` and `await1`) and maybe rename `enqueueOne` to `enqueue1`.

I suppose the user can also just do `dequeue.take(1)`.